app/src/sharedTest/java/org/oppia/app/testing/InputInteractionViewTestActivityTest.kt

```diff
@@ -673,4 +674,126 @@ class InputInteractionViewTestActivityTest {
     onView(withId(R.id.test_text_input_interaction_view)).check(matches(isDisplayed()))
       .check(matches(withText("abc")))
   }
+
+  @Test
+  fun testRatioInputInteractionView_withNoInputText_hasCorrectPendingAnswerType() {
+    val activityScenario = ActivityScenario.launch(
+      InputInteractionViewTestActivity::class.java
+    )
+    activityScenario.onActivity { activity ->
+      val pendingAnswer = activity.ratioExpressionInputInteractionViewModel.getPendingAnswer()
+      assertThat(pendingAnswer.answer.ratioExpression).isInstanceOf(RatioExpression::class.java)
```

Comment:
Suggest instead:

``assertThat(pendingAnswer.answer.objectTypeCase).isEqualTo(InteractionObject.ObjectTypeCase.RATIO_EXPRESSION)``

(you may need to use variables and/or extra imports to shorten this line). I think your current implementation will always pass because proto buff will return a default object when one is absent (though I'm not as sure about oneof behavior--it might throw). Checking for the type of answer should always be done via proto cases instead of object types.

Reply:
Done. Thanks for the clarification